mungegithub/publisher/cmd/collapsed-kube-commit-mapper/main.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/golang/glog"
+	gogit "gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+
+	"k8s.io/test-infra/mungegithub/publisher/pkg/cache"
+	"k8s.io/test-infra/mungegithub/publisher/pkg/git"
+)
+
+func Usage() {
+	fmt.Fprintf(os.Stderr, `Print a lookup table by printing each mainline k8s.io/kubernetes
+commit hash with its corresponding commit hash in the current branch
+(which is the result of a "git filter-branch --sub-directory"). It is
+expected that the commit messages on the current branch contain a
+"Kubernetes-commit: <upstream commit>" line for the directly corresonding
+commit. Note, that a number of k8s.io/kubernetes mainline commits might
```

Comment:
Are the collapsed mainline commits the fast-forward merges?

Reply:
All commits which do not have changes to the filter subdirectory. This can be ff-merges in k8s.io/kubernetes, but in general every commits which is empty after the filtering.